charts/pega/charts/installer/templates/pega-installer-job.yaml

```diff
@@ -4,15 +4,15 @@
 {{ if (and  (eq (include "performOnlyUpgrade" .) "true") (eq .Values.upgrade.upgradeType "in-place"))  }}
 {{ template "pega.installer" dict "root" $ "name" (include "pegaDBInPlaceUpgrade" .) "action" "upgrade" }}
 {{ end }}
-{{ if (eq (include "performUpgradeAndDeployment" .) "true") }}
+{{ if ( and (eq (include "performUpgradeAndDeployment" .) "true") (eq .Values.upgrade.upgradeType "zero-downtime")) }}
 {{ template "pega.installer" dict "root" $ "name" (include "pegaPreDBUpgrade" .) "action" "pre-upgrade" }}
 {{ template "pega.installer" dict "root" $ "name" (include "pegaDBZDTUpgrade" .) "action" "upgrade" "initContainers" (list "waitForPreDBUpgrade") }}
 {{ template "pega.installer" dict "root" $ "name" (include "pegaPostDBUpgrade" .) "action" "post-upgrade" "initContainers" (list "waitForPegaDBZDTUpgrade" "waitForRollingUpdates") }}
 {{ end }}
 {{ if ( and (eq (include "performOnlyUpgrade" .) "true") (eq .Values.upgrade.upgradeType "out-of-place")) }}
 {{ template "pega.installer" dict "root" $ "name" (include "pegaDBOOPUpgrade" .) "action" "upgrade" }}
 {{ end }}
-{{ if ( and  (eq (include "performOnlyUpgrade" .) "true") (eq .Values.upgrade.upgradeType "custom")) }}
+{{ if ( and  ( or (eq (include "performUpgradeAndDeployment" .) "true") (eq (include "performOnlyUpgrade" .) "true") ) (eq .Values.upgrade.upgradeType "custom")) }}
```

Comment:
Please add some comment here for future references - why this condition is changed and what purpose it serves.

Reply:
Since ZDT supports the resume feature in case of failure, we can switch to a custom upgrade and perform the rest of the installation steps while keeping the pods alive. To do this, we enabled the installer to run in case of `performUpgradeAndDeployment`